docs/source/developers/guide/resources.rst

```diff
@@ -27,3 +27,52 @@
 ************************************
 Additional information and resources
 ************************************
+
+On this page we will list resources that may be relevant or useful for 
+contributors who want to learn more about different parts of Apache Arrow.
+
+.. Annotation Glossary
+.. -------------------
+.. ARROW-15130
+
+Additional information
+----------------------
+
+- GitHub Actions
+
+  GitHub Actions is a continuous integration and continuous delivery (CI/CD) platform.
+  In Apache Arrow we use GitHub actions to build and test every pull request that is
+  opened, merged etc. To do so there are workflows created and then run via GitHub actions.
```

Comment:
* `GitHub actions` --> `GitHub Actions`
* `opened, merged etc.` --> `opened, merged, etc.`

Suggestion for merging 2nd/3rd sentences:
`Apache Arrow has a set of workflows that run via GitHub Actions to build and test every pull request that is opened, merged, etc.`